### PR TITLE
Update dependencies after parsec release + General Maintenance

### PR DESCRIPTION
**Pull request by tgonzalezorlandoarm**
- Update parsec-client and psa-crypto dependencies
- Start tracking Cargo.lock
- Use parsec 1.3.0 for testing

### CODE REVIEW

**Comment by tgonzalezorlandoarm** [comment]
```
+  // At the moment of writing, this would only fail with a wrongly configured endpoint. The
-   // We do not use "new" directly here because it can fail and we do not want to panic. The
    // automatic selection of authentication/provider is done in `p_init` so that we can return an
    // error if there is a problem with that.
    static ref PARSEC_BASIC_CLIENT: RwLock<BasicClient> = RwLock::new(BasicClient::new_naked().unwrap());
```

Now `new_naked()` does indeed return a Result. Looking at the [client code](https://github.com/parallaxsecond/parsec-client-rust/blob/0.16.0/src/core/basic_client.rs#L131) , it only throws an error when the parsec service endpoint is incorrectly defined, so we do want it to panic there. Does everyone agree that this is okay? If not, I'm open to suggestions :)

**Comment by gowthamsk-arm** [comment]
`// We do not use "new" directly here because it can fail and we do not want to panic.`
Do you know why this was done in the first place? Its unclear for me.

File: src/lib.rs [hunk 1]
```diff
@@ -57,10 +57,10 @@ use std::ptr;
 use std::sync::RwLock;
 
 lazy_static! {
-    // We do not use "new" directly here because it can fail and we do not want to panic. The
+    // At the moment of writing, this would only fail with a wrongly configured endpoint. The
     // automatic selection of authentication/provider is done in `p_init` so that we can return an
     // error if there is a problem with that.
-    static ref PARSEC_BASIC_CLIENT: RwLock<BasicClient> = RwLock::new(BasicClient::new_naked());
+    static ref PARSEC_BASIC_CLIENT: RwLock<BasicClient> = RwLock::new(BasicClient::new_naked()?);
```
Comment (gowthamsk-arm) [inline src/lib.rs]: Can we squash this change with the other commit ?

Comment (tgonzalezorlandoarm) [reply src/lib.rs]: I think your comment is marked as "outdated". Is this still valid? If so, can you specify which commits?

**Comment by tgonzalezorlandoarm** [comment]
> `// We do not use "new" directly here because it can fail and we do not want to panic.` Do you know why this was done in the first place? Its unclear for me.

So, `new_naked()` did not return a `Result` an error before, hence, there was no failure possibility. This way, the checks and potential failures only happened when `p_init` was called, which was a good thing as this would then be handled by the driver accordingly. Now, there is a potential for failure when initializing the client at the start but not through the driver.

File: src/lib.rs (hunk 1 above)
Comment (gowthamsk-arm) [reply src/lib.rs]: Its from this [commit](https://github.com/parallaxsecond/parsec-se-driver/pull/36/commits/ea8b797dee86a7d2aacf90a062570ea0238b0870)

**Comment by gowthamsk-arm** [comment]
From what I understand, `lazy_static` variables are only initialized when they are first accessed. (Please correct me if I'm wrong here) So either way, the potential failures should happen only when `p_init` is called right?

**Comment by tgonzalezorlandoarm** [comment]
> From what I understand, `lazy_static` variables are only initialized when they are first accessed. (Please correct me if I'm wrong here) So either way, the potential failures should happen only when `p_init` is called right?

you are right :)

File: src/lib.rs (hunk 1 above)
Comment (tgonzalezorlandoarm) [reply src/lib.rs]: Done

**Review by gowthamsk-arm — APPROVED** [review]
LGTM. Thanks.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -25,6 +25,7 @@ jobs:
         run: git clone -b v3.0.0 https://github.com/ARMmbed/mbedtls.git
       - name: armv7-unknown-linux-gnueabihf
         run: |
+          sudo apt update
```
Comment (ionut-arm) [inline .github/workflows/ci.yml:28]: Is the `sudo` here necessary?

File: src/lib.rs [hunk 3]
```diff
@@ -86,16 +86,26 @@ unsafe extern "C" fn p_init(
...
-        return PSA_ERROR_GENERIC_ERROR;
+            if let Err(e) = client.set_default_provider() {
+                error!("Error setting the default provider ({}).", e);
+                return PSA_ERROR_GENERIC_ERROR;
+            }
+        }
+        Err(e) => {
+            error!("Error Setting up the Parsec Client ({}).", e);
+            return PSA_ERROR_GENERIC_ERROR;
+        }
```
Comment (ionut-arm) [inline src/lib.rs:108]: Excellent, thanks for adding the error handling here!

**Review by ionut-arm — APPROVED** [review]
LGTM!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (tgonzalezorlandoarm) [reply .github/workflows/ci.yml:28]: I remember trying it and it was. Same for the `sudo apt install` s that come in the next lines that we already there

**Pull request merged**